tensorboard/webapp/widgets/line_chart_v2/lib/integration_test.ts

```diff
@@ -0,0 +1,197 @@
+      });
+      chart.updateViewBox({x: [0, 10], y: [0, 10]});
+      chart.updateData([createSeries('line1', (index) => index)]);
+
+      expect(dom.children.length).toBe(1);
+    });
+  });
+
+  describe('chart update', () => {
+    it('updates lines on data and metadata changes', () => {
+      chart.updateMetadata({
+        line1: buildMetadata({id: 'line1', visible: true}),
+        line2: buildMetadata({id: 'line2', visible: true}),
+      });
+      chart.updateViewBox({x: [0, 10], y: [0, 20]});
+      chart.updateData([createSeries('line1', (index) => index)]);
+
+      expect(dom.children.length).toBe(1);
+      const path1 = dom.children[0] as SVGPathElement;
+      expect(path1.getAttribute('d')).toBe(
+        'M0,100L20,95L40,90L60,85L80,80L100,75L120,70L140,65L160,60L180,55'
+      );
+      expect(path1.style.display).toBe('');
+
+      chart.updateData([
+        createSeries('line1', (index) => index),
+        createSeries('line2', (index) => index * 2),
+      ]);
+      expect(dom.children.length).toBe(2);
+      const path2 = dom.children[1] as SVGPathElement;
+      expect(path2.getAttribute('d')).toBe(
+        'M0,100L20,90L40,80L60,70L80,60L100,50L120,40L140,30L160,20L180,10'
+      );
+
+      chart.updateMetadata({
+        line1: buildMetadata({id: 'line1', visible: false}),
+        line2: buildMetadata({id: 'line2', visible: true}),
+      });
+
+      expect(dom.children.length).toBe(2);
+      const path1After = dom.children[0] as SVGPathElement;
+      expect(path1After.style.display).toBe('none');
+    });
+
+    it('updates on viewBox changes', () => {
+      chart.updateMetadata({
+        line: buildMetadata({id: 'line', visible: true}),
+      });
+      chart.updateViewBox({x: [0, 10], y: [0, 20]});
+      chart.updateData([createSeries('line', (index) => index)]);
+
+      const path = dom.children[0] as SVGPathElement;
+      expect(path.getAttribute('d')).toBe(
+        'M0,100L20,95L40,90L60,85L80,80L100,75L120,70L140,65L160,60L180,55'
+      );
+
+      chart.updateViewBox({x: [0, 10], y: [0, 10]});
+      expect(path.getAttribute('d')).toBe(
+        'M0,100L20,90L40,80L60,70L80,60L100,50L120,40L140,30L160,20L180,10'
+      );
+    });
+
+    it('updates once a requestAnimationFrame', async () => {
+      chart.updateMetadata({
+        line: buildMetadata({id: 'line', visible: true}),
+      });
+      chart.updateViewBox({x: [0, 10], y: [0, 100]});
+      chart.updateData([createSeries('line', (index) => index)]);
+
+      const path = dom.children[0] as SVGPathElement;
+      const pathDBefore = path.getAttribute('d');
+
+      const rafCallbacks: FrameRequestCallback[] = [];
+      rafSpy.and.callFake((cb: FrameRequestCallback) => {
+        return rafCallbacks.push(cb);
+      });
+
+      chart.updateViewBox({x: [0, 10], y: [0, 1]});
+
+      expect(rafCallbacks.length).toBe(1);
+      expect(path.getAttribute('d')).toBe(pathDBefore);
+
+      rafCallbacks.shift()!(1);
```

Comment:
Do we want to test that `onDrawEnd` is called properly?

Reply:
Added one.